### PR TITLE
Helper Buddy 3.0! (WIP)

### PR DESCRIPTION
**Pull request by TullyBurnalot**
Introducing... the brand new Helper Buddy 3.0!

(Help Buddies 1.0 through 2.7 discontinued due to eardrum damage. We fixed it, we swear)

- NanoTrasen-brand Helper Buddy Machine added to serve as a basic helper to all new arrivals to the station! Simply open the menu and select which Lesson you want to learn!

- NanoTrasen-brand Helper Buddy will not be a substitute for proper experience, but will serve as a quick introduction to Departments!

- NanoTrasen-brand Helper Buddy is fully constructible, and comes with genuine *ping action!

- Due to previous lawsuits, NanoTrasen-Brand Helper Buddy cannot produce more than one message at a time;

- Now with gloriously rendered NanoUI

Goals:

- Add Helper Buddy messages for all Departments;
 - Add Department Helper Buddies, to help newcomers know what Departments as a whole do;
- Map the Helper Buddies;
- Make the Helper Buddy generate a hologram when in use

:cl:
add: Adds Helper Buddy 3.0 machine. Helper Buddy allows any player to open its menu and request basic information on a multitde of subjects.
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
Travis disapproves

**Comment by TullyBurnalot** [comment]
GODDAMNIT TRAVIS

**Comment by FreeStylaLT** [comment]
>code/game/machinery/helperbuddy.dm:57:warning: if: if statement has no effect

**Comment by TullyBurnalot** [comment]
GODDAMNIT TULLY

**Comment by Fox-McCloud** [comment]
This seems completely and totally unnecessary.

**Comment by FlattestGuitar** [comment]
This is hilariously amazing. 👍

**Comment by TullyBurnalot** [comment]
> books

Literally never, ever seen anyone pick up a book outside of the library for fun. This thing comes pre-packaged with adequately-labeled instructions.

> wiki

As many people continuously point out, a multitude of players just don't use the wiki. This is a much more easily-accessible IC alternative that may actually get people to start asking more questions rather than just winging it and getting everyone slowly annoyed.

> mentorchat

See the above response.

> Don't need a machine blaring out things either, IMO.

Turning it into user-only messages is quite viable and would not change the intention of this, so I would not be averse to swapping it around.

File: code/game/machinery/helperbuddy.dm [hunk 1]
```diff
@@ -0,0 +1,217 @@
+/obj/machinery/helperbuddy
+  name = "Helper Buddy 3.0"
```
Comment (Fox-McCloud) [inline code/game/machinery/helperbuddy.dm]: this spacing. Tabs...not spaces.

**Comment by CrAzYPiLoT-SS13** [comment]
Seems like less advanced mentorchat to me. A bit useless, but I'm not opposed to it.

File: code/game/machinery/helperbuddy.dm (hunk 1 above)
Comment (TullyBurnalot) [inline code/game/machinery/helperbuddy.dm]: ... the hell. Fixed.

File: code/game/machinery/helperbuddy.dm [hunk 2]
```diff
@@ -0,0 +1,216 @@
+/obj/machinery/helperbuddy
+  name =  "Helper Buddy 3.0"
```
Comment (TheDZD) [inline code/game/machinery/helperbuddy.dm]: Why is this entire file indented with spaces instead of hard tabs?

**Comment by Lady-Luck** [comment]
I'm all for helping newbies in any way possible as long as it's not obtrusive. Pretty good ideas here.

**Comment by Fox-McCloud** [comment]
This would be better served as something that's a menu or verb or...something---not machines that are placed all over.

**Comment by Lady-Luck** [comment]
How about making it so helper buddies are an inventory item, and make it a 0 cost loadout item that new players have selected by default?

**Comment by tigercat2000** [comment]
that would require a snowflake system for loadouts
no

**Comment by Lady-Luck** [comment]
>that would require a snowflake system for loadouts

What?

**Comment by tigercat2000** [comment]
you would have to change the loadout system to make a "selected by default" item

**Comment by Lady-Luck** [comment]
>you would have to change the loadout system to make a "selected by default" item

So new features are snowflakes?

**Comment by Lady-Luck** [comment]
Whatever, it can easily just be a game preference that's on by default alongside the other game preferences. I just suggested the loadout system because it would be a neat way to make use of it.

**Comment by tigercat2000** [comment]
yes new features are snowflakes when you make them literally just for one item

**Comment by Lady-Luck** [comment]
>yes new features are snowflakes when you make them literally just for one item

It wouldn't have to be just one item. I'd love to see most default equipment for each job included in the loadout system and being able to choose not to spawn with certain items.

But that's just getting off-topic. A game preference would be just fine, really.

**Comment by Krausus** [comment]
> a multitude of players just don't use the wiki [or mentorchat]

Here's the thing: it doesn't matter if you have an out-of-game wiki, in-game tutorial books, mentors, tutorial machines, annoying tutorial fairies, tutorial popups, mandatory in-game tutorial sequences, *whatever,* some people simply refuse to learn a game proactively. The people who can't be arsed to read the wiki or an in-game book also can't be arsed to stand around listening to an in-game machine tell them what to do.

Plus, the existence of machines to teach people the basics of their job, on a station run by a company that presumably hires people capable of doing their job, is kinda out of place. It would make way more sense to have this sort of stuff as "reference guides" on players' PDAs... but if people won't read conveniently-located reference books, they probably wouldn't read those either.

**Comment by LittleBigKid2000** [comment]
DEY TUK ER JERBS
Also, I agree with Krausus. If they aren't going to use the wiki or a book or mentorhelp, they aren't going to use this either. And it breaks muh immersions.

**Comment by Aurorablade** [comment]
Well it gives them less of a I didn't know better excuse. I like the idea.

**Comment by IcyV** [comment]
>This seems completely and totally unnecessary. We have books, the wiki, and mentorchat.

**Comment by Vivalas** [comment]
To be honest I really don't like the aesthetics of this. It just seems out of place. NT's worker would be trained beforehand on their jobs, and this kinda feels like a hugbox. Makes it feel more like an MMO or something, dunno. It just doesn't fit in the SS13 environment. We already have so many resources available for new players, and if they're too lazy to utilize them, it's their funeral. I haven't looked at the changes so corecct me if I'm wrong but I'm guessing it's real basic information that everyone already knows other than new players. We also have that big examine system that nobody uses, I feel like that's a better use for this.

**Comment by HereticChurch** [comment]
I haven't played Goon in a goddamn long time, but don't they have a little robot guy who can show new players around? I'd support something like that. This is cool, but like everyone else has said, there are various methods for players to learn things already, and if they don't want to learn proactively, then that's their own fault.

**Comment by Earthdivine** [comment]
Kinda would prefer this to be a PDA app on the arrival tables. It's intrusive to the rest of the players and they can have it going in the back ground reading it while doing over things like walking around.

**Comment by Vivalas** [comment]
I like the idea here but don't think this is the proper way of doing this. It's too obtrusive and doesn't fit the atmosphere of the game. I like Earthdivine's idea as an option. Using the examine system that was made specifically for this would be cool too, or some kind of OOC (or IC pda app) called Robustipedia that has ingame information like this would be cool, kinda like on Civ or Spore.

File: code/game/machinery/helperbuddy.dm [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    sleep(50)
+    atom_say("<B>There are two main ways of ressuscitating a dead person if Cloning is either not a viable method, or is simply not available!</B>")
+    sleep(50)
+    atom_say("<B>The easiest method is the use of the Defibrillator. Make sure to wear it on your back, unhook the paddles, and aim for the person's chest. If they died recently, the Defibbrilator will be able to restart their heart!</B>")
+    sleep(50)
+    atom_say("<B>Keep in mind, this will not work if the person's body is severely damaged. Make sure to treat injuries prior to using the Defibrillator</B>")
+    sleep(50)
+    atom_say("<B>The harder method is via the use of Strange Reagent. This complex chemical can, if used in very small doses, raise someone from the dead!</B>")
+    sleep(50)
+    atom_say("<B>If dosage is exceeded, however, the person receiving the Strange Reagent will explode. Use with caution.</B>")
```
Comment (monster860) [inline code/game/machinery/helperbuddy.dm]: No, it will explode if there is mroe than 150 brute+burn da,age

**Comment by TullyBurnalot** [comment]
Following reception, closing this to come up with a better idea.

**Pull request closed without merging**